src/ai/backend/web/proxy.py

```diff
@@ -127,22 +127,33 @@ async def close_upstream(self) -> None:
             await self.up_conn.close()
 
 
-async def decrypt_payload(request: web.Request) -> str:
-    if not request.content:
-        return ""
-    config = request.app["config"]
-    scheme = config["service"]["force_endpoint_protocol"]
-    if scheme is None:
-        scheme = request.scheme
-    api_endpoint = f"{scheme}://{request.host}"
-    payload = await request.text()
-    iv, real_payload = payload.split(":")  # Extract initial vector and actual payload
-    key = (base64.b64encode(api_endpoint.encode("ascii")).decode() + iv + iv)[0:32]
-    crypt = AES.new(bytes(key, encoding="utf8"), AES.MODE_CBC, bytes(iv, encoding="utf8"))
-    b64p = base64.b64decode(real_payload)
-    dec = unpad(crypt.decrypt(bytes(b64p)), 16)
-    result = dec.decode("UTF-8")
-    return result
+@web.middleware
+async def decrypt_payload(request: web.Request, handler) -> web.StreamResponse:
+    request_headers = extra_config_headers.check(request.headers)
+    secure_context = request_headers.get("X-BackendAI-Encoded", None)
+    if secure_context:
+        if not request.content:
+            request.app["config"]["payload"] = ""
+            return await handler(request)
+        config = request.app["config"]
+        scheme = config["service"]["force_endpoint_protocol"]
+        if scheme is None:
+            scheme = request.scheme
+        api_endpoint = f"{scheme}://{request.host}"
+        payload = await request.text()
+        iv, real_payload = payload.split(":")  # Extract initial vector and actual payload
```

Comment:
I wonder if it would be right to place an additional comment here.
Maybe we can set the variable name to `initial_vector` rather than `iv` and remove the comment?

Reply:
Should I replace the existing `iv` with `initial_vector`?

Reply:
In my personal opinion, yes, if the comment is only for describing what `iv` is.

Reply:
When the variable name of iv is changed, it is thought that the variable name is too long when used in the next line. Would it be okay?

Reply:
I didn't think about that part because I used it without changing the logic of the existing decrypt_payload function. I think it would be better to change it like you said.